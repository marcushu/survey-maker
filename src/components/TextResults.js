import React from 'react';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import PropTypes from 'prop-types';


export default function TextResults({ surveyData }) {

  const textResponses = answerObj => {
    let textResponses = [];

    for (let k in answerObj) {
      if (k !== "question" && k !== "questionType" && answerObj.questionType === "TEXT")
        textResponses.push({ textAnswer: k, textQuestion: answerObj.question })
    }

    return textResponses;
  }


  return (
    <>
      {surveyData.map((res, index) =>
        <div key={index}>
          {res.questionType === "TEXT" &&
            <Row style={{ margin: "0px", padding: "5px" }}>
              <Col>
                <br/>
                <h5 style={{backgroundColor: "rgb(218, 218, 218)", padding: "5px"}}>
                  {res.question}
                </h5>
              </Col>
            </Row>
          }
          {textResponses(res).map(textObject =>
            <Row style={{ margin: "0px", padding: "5px" }}
              key={textObject.textAnswer}>
              <Col>
                <span style={{ fontStyle: "italic" }}>
                  "{textObject.textAnswer}"
                </span>
              </Col>
            </Row>
          )}
        </div>
      )}
    </>
  )
}

TextResults.propTypes = {
  surveyData: PropTypes.array
}