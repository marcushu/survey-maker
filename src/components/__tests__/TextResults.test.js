import React from 'react';
import { render } from '@testing-library/react';
import TextResults from '../TextResults';

describe("TextResults", () => {
    const mockData = [
        {
            No: 1,
            Yes: 3,
            question: "did you attend",
            questionType: "BINARY"
        },
        {
            "12 mostly family": 1,
            "2": 1,
            "3, myselfe, my wife and our dog": 1,
            asdfasfd: 1,
            question: "how many in your party",
            questionType: "TEXT"
        },
        {
            good: 1,
            great: 3,
            question: "the event was...",
            questionType: "MULTCHOICE"
        },
        {
            No: 2,
            Yes: 2,
            question: "I had fun.",
            questionType: "BINARY"
        }
    ];

    it("renders", () => {
        const { asFragment } = render(
            <TextResults surveyData={mockData} />
        );

        expect(asFragment()).toMatchSnapshot();
    });

    it("prints the question and answer", () => {
        const { getByRole, getByText } = render(
            <TextResults surveyData={mockData} />
        );

        expect(getByRole('heading', { name: /how many in your party/i })).toBeInTheDocument();
        expect(getByText(/"12 mostly family"/i)).toBeInTheDocument();
    })
})