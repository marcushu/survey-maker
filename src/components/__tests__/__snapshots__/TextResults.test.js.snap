// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextResults renders 1`] = `
<DocumentFragment>
  <div />
  <div>
    <div
      class="row"
      style="margin: 0px; padding: 5px;"
    >
      <div
        class="col"
      >
        <br />
        <h5
          style="background-color: rgb(218, 218, 218); padding: 5px;"
        >
          how many in your party
        </h5>
      </div>
    </div>
    <div
      class="row"
      style="margin: 0px; padding: 5px;"
    >
      <div
        class="col"
      >
        <span
          style="font-style: italic;"
        >
          "2"
        </span>
      </div>
    </div>
    <div
      class="row"
      style="margin: 0px; padding: 5px;"
    >
      <div
        class="col"
      >
        <span
          style="font-style: italic;"
        >
          "12 mostly family"
        </span>
      </div>
    </div>
    <div
      class="row"
      style="margin: 0px; padding: 5px;"
    >
      <div
        class="col"
      >
        <span
          style="font-style: italic;"
        >
          "3, myselfe, my wife and our dog"
        </span>
      </div>
    </div>
    <div
      class="row"
      style="margin: 0px; padding: 5px;"
    >
      <div
        class="col"
      >
        <span
          style="font-style: italic;"
        >
          "asdfasfd"
        </span>
      </div>
    </div>
  </div>
  <div />
  <div />
</DocumentFragment>
`;
